snuba/cli/subscriptions_executor.py

```diff
@@ -23,15 +23,24 @@
     "--dataset",
     "dataset_name",
     required=True,
-    type=click.Choice(["events", "transactions", "metrics"]),
+    type=click.Choice(["events", "transactions", "metrics", "generic_metrics"]),
     help="The dataset to target.",
 )
 @click.option(
     "--entity",
     "entity_names",
     required=True,
     multiple=True,
-    type=click.Choice(["events", "transactions", "metrics_counters", "metrics_sets"]),
+    type=click.Choice(
```

Comment:
Is there any reason for not picking up the full list of entities? 
If the entity does not support subscriptions the user gets an error immediately (nothing different than getting the error from click).
This way, instead, we always need to touch the CLI file when adding an entity, which we generally forget and likely cause another back and forth when trying to deploy in prod.

Reply:
I'm going to merge this just to get alerts unblocked, but maybe I'll follow up with a PR that allows all entity names and then gives a more clearer message for the assertion of `assert result_topic_spec is not None` on line 117/126 which I assume is the place this would die